src/tms_service.ts

```diff
@@ -0,0 +1,337 @@
+  private readonly _config: ITMSService;
+  private readonly _proxyPath: string;
+
+  _getAbsoluteUrl = (url: string) => {
+    if (/^https?:\/\//.test(url)) {
+      return url;
+    } else {
+      return toAbsoluteUrl(this._emsClient.getTileApiUrl(), url);
+    }
+  };
+
+  _getRasterStyleJson = _.once(
+    async (): Promise<EmsRasterStyle | undefined> => {
+      const rasterUrl = this._getStyleUrlForLocale('raster');
+      if (rasterUrl) {
+        const url = this._proxyPath + this._getAbsoluteUrl(rasterUrl);
+        return this._emsClient.getManifest(this._emsClient.extendUrlWithParams(url));
+      } else {
+        return;
+      }
+    }
+  );
+
+  _getVectorStyleJsonRaw = _.once(
+    async (): Promise<EmsVectorStyle | undefined> => {
+      const vectorUrl = this._getStyleUrlForLocale('vector');
+      if (vectorUrl) {
+        const url = this._proxyPath + this._getAbsoluteUrl(vectorUrl);
+        const vectorJson = await this._emsClient.getManifest(
+          this._emsClient.extendUrlWithParams(url)
+        );
+        return { ...vectorJson };
+      } else {
+        return;
+      }
+    }
+  );
+
+  _getVectorStyleJsonInlined = _.once(
+    async (): Promise<EmsVectorStyle | undefined> => {
+      const vectorJson = await this._getVectorStyleJsonRaw();
+      if (vectorJson) {
+        const inlinedSources: EmsVectorSources = {};
+        const { sources } = vectorJson;
+        for (const sourceName of Object.getOwnPropertyNames(sources)) {
+          const { url } = sources[sourceName];
+          const sourceUrl = this._proxyPath + this._getAbsoluteUrl(url);
+          const extendedUrl = this._emsClient.extendUrlWithParams(sourceUrl);
+          const sourceJson = await this._emsClient.getManifest(extendedUrl);
+
+          const extendedTileUrls = sourceJson.tiles.map((tileUrl: string) => {
+            const url = this._proxyPath + this._getAbsoluteUrl(tileUrl);
+            return this._emsClient.extendUrlWithParams(url);
+          });
+          // Override the attribution in the sources with the localized attribution
+          const htmlAttribution = await this.getHTMLAttribution();
+          inlinedSources[sourceName] = {
+            type: 'vector',
+            ...sourceJson,
+            attribution: htmlAttribution,
+            tiles: extendedTileUrls,
+          };
+        }
+        return {
+          ...vectorJson,
+          sources: inlinedSources,
+          sprite: await this._getSpriteSheetRootPath(),
+          glyphs: await this._getUrlTemplateForGlyphs(),
+        };
+      } else {
+        return;
+      }
+    }
+  );
+
+  constructor(config: ITMSService, emsClient: EMSClient, proxyPath: string) {
+    this._config = config;
+    this._emsClient = emsClient;
+    this._proxyPath = proxyPath;
+  }
+
+  _getFormats(formatType: string, locale: string) {
+    return this._config.formats.filter(
+      format => format.locale === locale && format.format === formatType
+    );
+  }
```

Comment:
There are in this class a few methods that don't declare their return type. It's OK since they are inferred by the code but having them being explicitly declared would help in future refactorings.